packages/calcite-components/src/components/sheet/sheet.tsx

```diff
@@ -0,0 +1,365 @@
+  @Event({ cancelable: false }) calciteSheetOpen: EventEmitter<void>;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Public Methods
+  //
+  //--------------------------------------------------------------------------
+
+  /**
+   * Sets focus on the component's "close" button (the first focusable item).
+   *
+   */
+  @Method()
+  async setFocus(): Promise<void> {
+    await componentFocusable(this);
+    focusFirstTabbable(this.el);
+  }
+
+  /**
+   * Updates the element(s) that are used within the focus-trap of the component.
+   */
+  @Method()
+  async updateFocusTrapElements(): Promise<void> {
+    updateFocusTrapElements(this);
+  }
+
+  //--------------------------------------------------------------------------
+  //
+  //  Private Methods
+  //
+  //--------------------------------------------------------------------------
+
+  onBeforeOpen(): void {
+    this.calciteSheetBeforeOpen.emit();
+  }
+
+  onOpen(): void {
+    this.calciteSheetOpen.emit();
+    activateFocusTrap(this);
+  }
+
+  onBeforeClose(): void {
+    this.calciteSheetBeforeClose.emit();
+  }
+
+  onClose(): void {
+    this.calciteSheetClose.emit();
+    deactivateFocusTrap(this);
+  }
+
+  private setTransitionEl = (el: HTMLDivElement): void => {
+    this.transitionEl = el;
+    this.contentId = ensureId(el);
+  };
+
+  private openEnd = (): void => {
+    this.setFocus();
+    this.el.removeEventListener("calciteSheetOpen", this.openEnd);
+  };
+
+  private openSheet() {
+    if (this.ignoreOpenChange) {
+      return;
+    }
+
+    this.el.addEventListener("calciteSheetOpen", this.openEnd);
+    this.open = true;
+    this.opened = true;
+    if (!this.slottedInShell) {
+      this.initialOverflowCSS = document.documentElement.style.overflow;
+      // use an inline style instead of a utility class to avoid global class declarations.
+      document.documentElement.style.setProperty("overflow", "hidden");
+    }
+
+    this.ignoreOpenChange = false;
+  }
+
+  private handleOutsideClose = (): void => {
+    if (this.outsideCloseDisabled) {
+      return;
+    }
+
+    this.closeSheet();
+  };
+
+  private closeSheet = async (): Promise<void> => {
+    if (this.ignoreOpenChange) {
+      return;
+    }
+
+    if (this.beforeClose) {
+      try {
+        await this.beforeClose(this.el);
+      } catch (_error) {
+        // close prevented
+        requestAnimationFrame(() => {
+          this.ignoreOpenChange = true;
+          this.open = true;
+          this.ignoreOpenChange = false;
+        });
+        return;
+      }
+    }
+
+    this.ignoreOpenChange = true;
+    this.open = false;
+    this.opened = false;
+    this.removeOverflowHiddenClass();
+    this.ignoreOpenChange = false;
+  };
+
+  private removeOverflowHiddenClass(): void {
+    document.documentElement.style.setProperty("overflow", this.initialOverflowCSS);
+  }
+
+  private handleMutationObserver = (): void => {
```

Comment:
We don't need to bind this, so `handleMutationObserver` can be a simple method instead of an assigned, bound function.

```ts
private handleMutationObserver(): void {
  // ...
}
```

Applies to similar component members.

Reply:
applies to modal as well.